### DISCUSSION TITLE
Moving files in a GitHub repository without moving them on my computer

### DISCUSSION THREAD

**Discussion by s3862471** [community/community#23756] [New to GitHub]
<p>I created a 2nd GitHub project in it’s repository in GitHub, but it copied my file structure across so now the project (a web site) is buried in several levels of folders. I want to move all the files to the top level of the repository, but when I looked up moving files it said I should move them on my own computer. I don’t want to move the files on my computer, just in the GitHub repository.</p>
<p>It does work, it’s just a long URL at the moment. If I can do this from the command line interface it would be best as there are images involved, and I want to become more familiar with the CLI.</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi JoeKM888! :wave:  Moving files in the GitHub web UI is a bit of a pain. You can move files one by one by editing them and then editing the file path:</p>
<p><a href="https://help.github.com/en/github/managing-files-in-a-repository/moving-a-file-to-a-new-location" class="onebox" target="_blank">https://help.github.com/en/github/managing-files-in-a-repository/moving-a-file-to-a-new-location</a></p>
<p>That won’t work on some types of files that can’t be edited in the UI, such as images or PDFs, and in any case, moving them one by one is not very practical!</p>
<p>So the solution is indeed to move the folder on your local copy, and then push the change to GitHub. (This is how using the CLI to move things works too - you move you local copy, then push the change.)</p>
<p>Here’s a helpful article that explains how to use the <code>mv</code> command, which is how you move files and folders around on the command line in Unix type systems:</p>
<aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/19803303/181079238-e948126f-b188-4218-a5c7-1b7c642e31df.png" class="site-icon" width="32" height="32">
      <a href="https://shapeshed.com/unix-mv/" target="_blank">George Ornbo</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:583/118;"><img src="https://user-images.githubusercontent.com/19803303/181079240-b4fbb08c-f11b-4983-bfb3-8ecbc2fe2548.png" class="thumbnail" width="583" height="118"></div>

<h3><a href="https://shapeshed.com/unix-mv/" target="_blank">Linux and Unix mv command tutorial with examples | George Ornbo</a></h3>

<p>Tutorial on using mv, a UNIX and Linux command to move or rename files. Examples of moving a file, moving multiple files, moving a directory, prompting before overwriting and taking a backup before moving.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>(If you’re a Windows user, I recommend you download and use Git Bash for Windows:</p>
<p><a href="https://gitforwindows.org/" class="onebox" target="_blank">https://gitforwindows.org/</a></p>
<p>It uses Unix style syntax for commands, and you will find that almost all tutorials, blog posts, and stack overflow answers use the same syntax, so it will make everything easier in the future.)</p>
<p>After you’ve moved the folder, here’s how to commit and push the change:</p>
<p><a href="https://help.github.com/en/github/managing-files-in-a-repository/moving-a-file-to-a-new-location-using-the-command-line" class="onebox" target="_blank">https://help.github.com/en/github/managing-files-in-a-repository/moving-a-file-to-a-new-location-using-the-command-line</a></p>
